### PR TITLE
Super Circuit Invalid Precompile Inputs Integration Tests (OOG, EC Fp Overflow, EC Not on Curve)

### LINKED ISSUE #834: feat: begin_tx precompile out of gas error
@darth-cy .   need this test for super circuit

### PR DESCRIPTION
**Pull request by darth-cy**
### Description

Add additional precompile integration test on `super_circuit` for OOG, EC pts coordinates `Fp` overflow and EC pts not on curve. Refactor `super_circuit` precompile testing block trace generation. Follow up QA on #811.

### Issue Link

Closes #834 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by lispc** [comment]
@roynalnaruto do you think it is a good idea to use several super circuit circuits?   The benefit is when some precompiles fails, we can know which fails from ci quickly.  The drawback is it takes longer CI time.

I would slightly prefer to put multi txs into same super circuit. So CI will not be slower.  Usually it will not fail.  Even if it fails, it is not difficult to debug which one is the problem.

**Comment by lispc** [comment]
@darth-cy some clippy and fmt error

**Comment by roynalnaruto** [comment]
> do you think it is a good idea to use several super circuit circuits

cc @darth-cy I would prefer what @lispc mentioned above. In fact we don't even need multiple txs. We can have a single tx, calling a contract that does multiple `ec` operations, for static and non-static calls. It would take a bit of refactoring, but yes, it saves a ton of overhead by having a single circuit instance (hence a single test). Instead we can add trace logs at appropriate places so that we identify the failing case (ec op and EVM inputs for it)

**Comment by darth-cy** [comment]
@roynalnaruto I can bundle up the bytecode for multiple ec operations onto a single contract account, then it will just take 1 tx in most cases. Using this approach, OOG will still have separate txs and ec pairing cases unfortunately still have to be split onto ~~different txs~~ different tests (a new super circuit) due to rate limit. Should I modify the code for this approach?

**Comment by darth-cy** [comment]
@lispc cc @roynalnaruto The tests are now refactored and compacted into two tests (using 2 super circuit instances). An ec pairing test is done in ~~a separate~~ the second test due to rate limit on ec pairing. All tests pass and traces will be published.

File: eth-types/src/evm_types.rs [hunk 1]
```diff
@@ -200,7 +200,7 @@ impl GasCost {
     /// Per-pair gas cost for BN256 pairing
     pub const PRECOMPILE_BN256PAIRING_PER_PAIR: Self = Self(34_000);
     /// Base gas cost for precompile call: MODEXP
-    pub const PRECOMPILE_MODEXP: Self = Self(0);
+    pub const PRECOMPILE_MODEXP: Self = Self(200); // eip2565
```
Comment (lispc) [inline eth-types/src/evm_types.rs:203]: i think this change is correct.  Then why the old codes worked well..? @roynalnaruto

**Comment by lispc** [comment]
i think this case is not covered here? https://github.com/scroll-tech/zkevm-circuits/issues/834 cannot find it.

**Comment by z2trillion** [comment]
I marked this as ready for review only to trigger the CI.

**Comment by kunxian-xia** [comment]
mark ready to get ci up.

**Comment by kunxian-xia** [comment]
@z2trillion https://github.com/scroll-tech/zkevm-circuits/pull/850/commits/a33114399216ef8483ded33e57f176b3fb4b5a4e should be reverted as the "tx-circuit" related CI errors have already been fixed in https://github.com/scroll-tech/zkevm-circuits/pull/935.

File: eth-types/src/evm_types.rs (hunk 1 above)
Comment (roynalnaruto) [reply eth-types/src/evm_types.rs:203]: I think we were using `PRECOMPILE_MODEXP_MIN` which is defined below
